climada/hazard/tc_tracks.py

```diff
@@ -263,28 +291,82 @@ def tracks_in_exp(self, exposure, buffer=1.0):
+                            estimate_missing=False, correct_pres=False,
                             file_name='IBTrACS.ALL.v04r00.nc'):
         """Read track data from IBTrACS databse.
 
+        When using data from IBTrACS, make sure to be familiar with the scope and limitations of
+        IBTrACS, e.g. by reading the official documentation
+        (https://www.ncdc.noaa.gov/ibtracs/pdf/IBTrACS_version4_Technical_Details.pdf). Reading the
+        CLIMADA documentation can't replace a thorough understanding of the underlying data. This
+        function only provides a (hopefully useful) interface for the data input, but cannot
+        provide any guidance or make recommendations about if and how to use IBTrACS data for your
+        particular project.
+
+        Resulting tracks are required to have both pressure and wind speed information at all time
+        steps. Therefore, all track positions where one of wind speed or pressure are missing are
+        discarded unless one of `interpolate_missing` or `estimate_missing` are active.
+
         Some corrections are automatically applied, such as: `environmental_pressure` is enforced
         to be larger than `central_pressure`.
 
         Parameters
         ----------
-        provider : str, optional
-            If specified, enforce use of specific agency, such as "usa", "newdelhi", "bom", "cma",
-            "tokyo". Default: None (and automatic choice).
+        provider : str or list of str, optional
+            Either specify an agency, such as "usa", "newdelhi", "bom", "cma", "tokyo", or the
+            special values "official" and "official_3h":
+            * "official" means using the (usually 6-hourly) officially reported values of the
+              officially responsible agencies.
+            * "official_3h" means to include (inofficial) 3-hourly data of the officially
+              responsible agencies (whenever available).
+            If you want to restrict to the officially reported values by the officially responsible
+            agencies (`provider="official"`) without any modifications to the original official
+            data, make sure to also set `estimate_missing=False` and `interpolate_missing=False`.
+            Otherwise, gaps in the official reporting will be filled using interpolation and/or
+            statistical estimation procedures (see below).
+            If a list is given, the following logic is applied: For each storm, the variables that
+            are not reported by the first agency for this storm are taken from the next agency in
+            the list that did report this variable for this storm. For different storms, the same
+            variable might be taken from different agencies.
+            Default: ['official_3h', 'usa', 'tokyo', 'newdelhi', 'reunion', 'bom', 'nadi',
+            'wellington', 'cma', 'hko', 'ds824', 'td9636', 'td9635', 'neumann', 'mlc']
```

Comment:
Sorry to insist here, but I think this is because my first comment was not clear enough. I hope I can express myself better here. This again might not be needed in the docstring, but it is for clarification.
What time step (6 or 3 or whatever) is used when a list of agencies is used that does not contain 'official' or 'official_3h' ? Is it the one reported by the particular agency that has the information for this variable? So, can one end up with different reporint times for different variables for the same track?

Reply:
Yes, in IBTrACS, you have different reporting times for different variables for the same track. But that's not a problem, because all variables have the same fixed time dimension of length 360. For a fixed storm, the meaning of each entry in the time dimension has a fixed meaning for all variables associated with that storm. In other words: For each storm, there is a list of up to 360 (ordered) timestamps for which ANY agency reported ANY variable for that storm. NaNs are used to indicate that a particular variable hasn't been reported by a particular agency for a particular storm at a particular time step.

Our function `read_ibtracs_netcdf` first collects all physical variables (lat, lon, central pressure, wind speed, radius of maximum winds, storm radius, environmental pressure) of length 360 for each storm according to the specified `provider`-parameter. Next, it tries to fill all NaN values according to `interpolate_missing` and/or `estimate_missing`. After that, all remaining time steps are discarded for which one of eye position, central pressure or wind speed is still NaN.

Reply:
I included a note in my latest commit:
https://github.com/CLIMADA-project/climada_python/blob/6cb1ee7114d64d8f330a7f9e216506530b9b41ed/climada/hazard/tc_tracks.py#L311-L313